contrib/abcl-jvm-stepper/README.markdown

```diff
@@ -0,0 +1,441 @@
+ABCL-JVM-STEPPER
+================
+
+ABCL-JVM-STEPPER provides a prototype implementation of a stepper as a replacement for the empty cl:step.
+
+It can work on both the plain REPL and Slime
+
+This stepper is designed to work on compiled code unlike ABCL-STEPPER, which works on interpreted code.
+You need to instrument first the code before calling the step macro.
+
+It shares all features of ABCL-STEPPER with the exception that it cannot show values of lexicals variables because they
+are eliminated after the compilation and it can only show global variable values, this also affects to the features :inspect
+and :watch.The feature :globals allow to show the values of the bound symbols linked to special variables in the current package.
+
+
+Comparison with ABCL-STEPPER:
+-----------------------------
+
+Advantages:
+
+- Significantly faster
+
+- Completely isolated at thread level which means you can run multiple stepping sessions in different connections without any interference
+between them, with the main thread running as well. This is useful for production environments.
+
+- Can see the values of the arguments to the called functions out of the box
+
+- Works also on Sly
+
+
+Limitations:
+
+- The stepper allows to jump between function calls after instrumenting the related functions but it can't yet step into their non function calls subforms. A workaround to solve this issue is the use of the utility macro 'stackify' (or something similar) wrapping the selected form to analyze before the instrumentation and the stepper will stop in the form and the user will be able to see the output of the form.
+See how to do it in the example session below.
+
+```
+(defmacro stackify (form)
+  (let ((inner-stack-name
+          (gensym
+           (format nil "~a -> form#"
+                   (with-output-to-string (s) (pprint form s))))))
+    `(funcall (sys:fset ',inner-stack-name (lambda () ,form)))))
+```
+
+- It can't yet print intermediate multiple values, the first value will be printed instead
+
+
+Usage:
+-------
+
+The contrib exports the following symbols:
+
+jstepper:instrument-compile-function, jstepper:instrument-compile-call, jstepper:instrument-compile-form & jstepper:step
+
+The firsts 3 ones are used, as their names indicates for instrument the code when compiling to be ready for the stepper
+
+The last one is the implementation of cl:step macro.
+
+Attaching a sample session to illustrate the use of the stepper
+
+```
+CL-USER(1): (require :asdf)
+NIL
+CL-USER(2): (require :abcl-contrib)
```

Comment:
After the `require` of `asdf`, it is better style to use `asdf` to load things:
```
(asdf:load-system :abcl-contrib)
(asdf:load-system :abcl-jvm-stepper)
```

This communicates that the components are encapsulated as `asdf` system, therefore expectations like having an `asd` manifest for source components, be able to `M-. slime-browse-system SYSTEM` work are suggested.

Reply:
@easye I just updated the Readme using _asdf:load-system_ instead of _require_ to load the contrib